code/go/0chain.net/smartcontract/dbs/event/user_test.go

```diff
@@ -213,21 +213,16 @@ func TestAddAndUpdateStakePoolRewards(t *testing.T) {
 	defer closeDB()
 
 	// create new users
-	miners := make([]Miner, 10)
+	var prs []ProviderRewards
 	for i := 0; i < 10; i++ {
-		miners[i] = Miner{
-			Provider: Provider{
-				ID: fmt.Sprintf("m_%v", i),
-				Rewards: ProviderRewards{
-					ProviderID:   fmt.Sprintf("m_%v", i),
-					Rewards:      currency.Coin((i + 1) * 10),
-					TotalRewards: currency.Coin((i + 1) * 1000),
-				},
-			},
-		}
+		prs = append(prs, ProviderRewards{
+			ProviderID:   fmt.Sprintf("m_%v", i),
+			Rewards:      currency.Coin((i + 1) * 10),
+			TotalRewards: currency.Coin((i + 1) * 1000),
+		})
 	}
 
-	err := rewardProvider(eventDb, "miner_id", "miners", miners)
+	err := eventDb.rewardProviders(prs)
 	require.NoError(t, err)
```

Comment:
We need to check if the columns are truly updated

Reply:
Removed as this is a duplicate of the tests in stakepool_test.go, [TestEventDb_rewardProviders](https://github.com/0chain/0chain/blob/last_round/code/go/0chain.net/smartcontract/dbs/event/stakepool_test.go#L21) and [TestEventDb_rewardProviderDelegates](https://github.com/0chain/0chain/blob/last_round/code/go/0chain.net/smartcontract/dbs/event/stakepool_test.go#L189).